### DISCUSSION TITLE
[v7] Static navigation from a Native Stack inside a Drawer with TypeScript

### DISCUSSION THREAD

**Discussion by Javi** [react-navigation/react-navigation#12306] [Q&A]
Hi, I'm using the new static navigation layout in mobile app to manage appointments and my (simplified) structure is currently like the following: 

- a root native stack with a 
  - splash screen
  - login screen 
  - and a logged in drawer screen which contains
    - a home screen
    - a native stack for appointments with
      1 - a list of appointments
      2 - an appointment's details (with people)
      3 - a person's details
    - a native stack for contacts with
      1 - a list of people
      2 - a person's details (with appointments)
      3 - an appointment's details

So basically the main use of this app is traversing stacks in a drawer, and I'd like to be able to reuse detail screens and navigate between them in both stacks. Now, this is all laid out with v7's static creators as shown above, and TypeScript, and I'm also using the `useNavigation()` hook, trying to navigate from a list screen to a detail screen, having setup the global `RootParamList` as stated in the docs with `StaticParamList<typeof RootStack>`, yet the `.navigate()` function from the list screens will only accept the Root destinations, not the detail screen destinations. 

Is such a structure inherently wrong? Should I do something else like flatten all screens in a single stack like in the examples? Should I join all stacks in the drawer in the global type declaration? Something else? It could be worthwile adding an example to the repo for a complex case like this, I can make a PR if we figure it out. Thanks for reading this far! 🙏

**Comment by satya164** [comment]
You need to use the API for navigating to a screen in a nested navigator https://reactnavigation.org/docs/nesting-navigators/#navigating-to-a-screen-in-a-nested-navigator

Also see https://reactnavigation.org/docs/nesting-navigators/#best-practices-when-nesting

**Reply by Javi** [reply]
Yes, I'd read all of it, but I didn't want to navigate to the other screen's route but rather push the same screen on the existing route. I guess I'll have to flatten everything onto a single Stack and use that one several times in the drawer, and add that Stack to the global typing.
